JoinColony/colonyCDapp
# Fix: Handle uninstalled Staged Expenditure extension

## Description

This PR fixes an issue where users could interact with staged payments created using a different installation of the Staged Expenditure extension. In order for the checks to work properly, there is a [block-ingestor PR](https://github.com/JoinColony/block-ingestor/pull/285) that facilitates storing the relevant extension address on the expenditure object.

In addition, it fixes an issue with the flow where you could not see previous funding actions after you uninstalled the extension. That meant if there was an unclaimed motion, there was no way for you to claim it. 

## Testing

1. Install Staged Expenditure extension.
2. Create 4 staged payments - you might find the Redo action option useful with that:

<img width="374" alt="image" src="https://github.com/user-attachments/assets/e107c96d-ef0a-46ee-b343-4e70475fa448">

3. Leave the first payment at the review step. Advance each of the remaining payments to further steps:
 - Review
 - Funding
 - Release
 - Payment

4. Uninstall Staged Expenditure extension.

5. Verify that the uninstalled extension warning is shown in each payment you created. You should also be able to go back to any of the previous steps and see the details of the action related to that step (ie. the creator, date, etc.).

7. Now install Staged Expenditure extension again.

8. Verify the warning is still shown for payments created using the uninstalled extension. 

9. Create another Staged Expenditure payment and move it through all stages to check for regressions.

<img width="385" alt="image" src="https://github.com/user-attachments/assets/7c81699f-ae69-4fe0-bd56-e582317fc5b2">


<img width="381" alt="image" src="https://github.com/user-attachments/assets/7d6a3ac0-2e13-4dc4-8643-a45bf3745303">


<img width="386" alt="image" src="https://github.com/user-attachments/assets/395a2f0c-378a-43b1-9c42-0fc7ed05dd34">

<img width="384" alt="Screenshot 2024-10-20 at 22 49 19" src="https://github.com/user-attachments/assets/6615650c-3ce9-4d12-ad9a-6630d304e78f">

Resolves #3304
